docs/data/charts/gauge/gauge.md

```diff
@@ -1,15 +1,139 @@
+
+By default it's assumed to be between 0 and 100.
+You can modify this range with props `valueMin` and `valueMax`.
+
+{{"demo": "GaugeValueRangeNoSnap.js"}}
+
+## Arcs configuration
+
+You can modify the arc shape with the following props:
+
+- `startAngle`, `endAngle`: Angle range provided in degrees
+- `innerRadius`, `outerRadius`: Radius of the arc. It can be a number for fix number of px. Or a percentage string which will be a percent of the maximal available radius.
+- `cornerRadius`: It can be a number for fix number of px. Or a percentage string which will be a percent of distance between inner and outer radius.
+
+{{"demo": "ArcPlaygroundNoSnap.js"}}
+
+:::info
+Notice that the arc position is computed to let the Gauge take as much space as possible in the drawing area.
+
+Provide props `cx` and/or `cy` to fix the coordinate of the arc center.
 :::
+
+## Text configuration
+
+By default, the Gauge displays the value in the center of the arc.
+To modify it, use the `text` prop.
+
+This prop can be a string, or a formatter.
+In the second case, the formatter argument contains the `value`, `valueMin` and `valueMax`.
+
+To modify the layout of the text, use the class name `gaugeClasses.valueText`.
+
+{{"demo": "TextPlaygroundNoSnap.js"}}
+
+## Arc design
+
+You can customize the Gauge style with usual selectors.
+The `gaugeClasses` provides class names to select parts of the component, such as `valueText`, `valueArc`, and `referenceArc`.
+
+{{"demo": "ArcDesign.js"}}
+
+## Adding elements
+
+### Using the default Gauge
+
+If the default Gauge does not contain all the visual elements you're looking for, you can add them.
+THe first solution is to add your component as children.
+They will be stacked on top of the default rendering.
+
+```tsx
+import { Gauge } from '@mui/x-charts/Gauge';
+
+<Gauge value={25} valueMax={50}>
+  <AddedElement />
+</Gauge>;
+```
+
+### Using the Gauge container
+
+The second solution is to start from scratch with our components:
+
+- GaugeReferenceArc
+- GaugeValueArc
+- GaugeValueText
+
+```tsx
+import {
+  GaugeContainer,
+  Gauge,
+  GaugeReferenceArc,
+  GaugeValueArc,
+} from '@mui/x-charts/Gauge';
+
+<GaugeContainer value={25} valueMax={50}>
+  <GaugeReferenceArc />
+  <GaugeValueArc />
+  <AddedElement />
+</GaugeContainer>;
+```
+
+### Creating your components
+
+To create your own components, use `useGaugeState` hook which provides all you need about the gauge configuration:
+
+- Information about the value: `value`, `valueMin`, `valueMax`
+- Information to plot the arc: `startAngle`, `endAngle`, `outerRadius`, `innerRadius`, `cornerRadius`, `cx`, and `cy`
+- Computed values:
+  - `maxRadius` the maximal radius that can fit in the drawing area.
+  - `valueAngle` the angle associated to the current value.
+
+{{"demo": "CompositionExample.js"}}
+
+## Accessibility
+
+(WAI-ARIA: [https://www.w3.org/WAI/ARIA/apg/patterns/meter/](https://www.w3.org/WAI/ARIA/apg/patterns/meter/))
```

Comment:
```suggestion
The MUI X Gauge chart is compliant with the [Meter ARIA pattern](https://www.w3.org/WAI/ARIA/apg/patterns/meter/), which includes the addition of the `meter` role to the parent container and correct usage of the `aria-valuenow`, `aria-valuemin`, and `aria-valuemax` attributes.
```

I think that we can be way more descriptive regarding the Accessibility section. Adding just the link to the ARIA docs doesn't feel super polished, and the reader should be able to get the main gist of it without navigating way from our docs!

Reply:
That was mostly inspired by the material-ui docs, but I agree :+1:

https://mui.com/material-ui/react-autocomplete/#accessibility